src/Scanner.js

```diff
@@ -233,7 +239,7 @@ class DefaultScanner {
         let traversalLayerCount = 0;
         let element = input;
         // traverse the DOM to search for related inputs
-        while (traversalLayerCount <= 5 && element.parentElement !== document.documentElement) {
+        while (traversalLayerCount <= 10 && element.parentElement !== document.documentElement) {
```

Comment:
This is a compromise, although I don't see any big issues at least in the test suite. Let me know if you think otherwise @GioSensation.